Source/DafnyLanguageServer.Test/Synchronization/CachingTest.cs

```diff
@@ -26,6 +28,64 @@ public class CachingTest : ClientBasedLanguageServerTest {
     serverOptions.Services.Replace(new ServiceDescriptor(typeof(ILoggerFactory), factory));
   }
 
+  [Fact]
+  public async Task ChangedImportAffectsExport() {
+
+    var importedSource = @"
+module Imported {
+  const x: int := 3
+  class Foo {
+    const r := 2
+
+    constructor() {
+
+    }
+  }
+}
+".TrimStart();
+
+    var exporter = @"
+module Exporter {
+  import Imported
+  export provides Imported, FooAlias, Wrapper
+
+  class Wrapper<T> {
+    method V() returns (r: T)
+  }
+  type FooAlias = Wrapper<Imported.Foo>
+}
+".TrimStart();
+
+    var importerSource = @"
+module Importer {
+  import Exporter
+  const i: int := 3
+  const x := Exporter.Imported.x
+
+  method Faz() {
+    var z : Exporter.FooAlias;
+
+    var q := new Exporter.Imported.Foo();
+    print q.r;
+  }  
+}
+".TrimStart();
+
+    var directory = Path.GetRandomFileName();
+    await CreateAndOpenTestDocument("", Path.Combine(directory, DafnyProject.FileName));
+    var imported = await CreateAndOpenTestDocument(importedSource, Path.Combine(directory, "imported.dfy"));
+    await CreateAndOpenTestDocument(exporter, Path.Combine(directory, "exporter.dfy"));
+    var importer = await CreateAndOpenTestDocument(importerSource, Path.Combine(directory, "importer.dfy"));
+
+    var diagnostics1 = await diagnosticsReceiver.AwaitNextDiagnosticsAsync(CancellationToken);
+    Assert.Empty(diagnostics1.Where(d => d.Severity == DiagnosticSeverity.Error));
+    ApplyChange(ref imported, ((0, 0), (0, 0)), "//comment" + Environment.NewLine);
+    ApplyChange(ref importer, ((2, 18), (2, 19)), "true");
+    var diagnostics2 = await diagnosticsReceiver.AwaitNextDiagnosticsAsync(CancellationToken);
+    Assert.Single(diagnostics2.Where(d => d.Severity == DiagnosticSeverity.Error));
```

Comment:
The error was not reported because of an internal exception, and that occurred because the `ExportModuleDecl` was retrieved from a cache when it should not have, which caused its references to be outdated, which caused a dictionary lookup failure.